### PR TITLE
Adds supports for AWS EKS

### PR DESCRIPTION
**Pull request by anneschuth**
I was missing support for connecting to AWS EKS, this PR adds support.

### CODE REVIEW

**Comment by anneschuth** [comment]
@superbrothers, curious to hear what you think!

File: assets/common.sh [hunk 1]
```diff
@@ -79,6 +79,26 @@ setup_kubectl() {
...
     exe kubectl config use-context "$CONTEXT_NAME"
+
+    # Optional. Use the AWS EKS authenticator
+    local use_aws_iam_authenticator
+    use_aws_iam_authenticator="$(jq -r '.source.use_aws_iam_authenticator // ""' < "$payload")"
+    local aws_eks_cluster_name
+    aws_eks_cluster_name="$(jq -r '.source.aws_eks_cluster_name // ""' < "$payload")"
+    if [[ "$use_aws_iam_authenticator" == "true" ]]; then
+      if [ -z "$aws_eks_cluster_name" ]; then
+        echoerr 'You must specify a clustername when using aws_iam_authenticator.'
```
Comment (JeroenJansze2018) [inline assets/common.sh]: Can we make this more precise? with file/variable that needs to be changed or an example even

File: Dockerfile [hunk 2]
```diff
@@ -6,13 +6,27 @@ ARG KUBERNETES_VERSION=
 
 RUN set -x && \
     apt-get update && \
-    apt-get install -y jq curl && \
+    apt-get install -y jq curl git && \
```
Comment (anneschuth) [inline Dockerfile]: Surprisingly (?) `go get` requires git.

**Comment by superbrothers** [comment]
Thank you for PR. To proceed to merge this PR, please read our [CLA](https://zlabjp.github.io/cla/). If you accept it, please say so in this thread.

**Comment by anneschuth** [comment]
Thanks @superbrothers, I accept the CLA.

**Comment by superbrothers** [comment]
Did you try using this resources built from your PR? Did it work correct? I've never used EKS, so I don't know how aws-iam-authenticator get token. It looks like that token isn't contained in kubeconfig file.

```bash
    # Optional. Use the AWS EKS authenticator
    local use_aws_iam_authenticator
    use_aws_iam_authenticator="$(jq -r '.source.use_aws_iam_authenticator // ""' < "$payload")"
    local aws_eks_cluster_name
    aws_eks_cluster_name="$(jq -r '.source.aws_eks_cluster_name // ""' < "$payload")"
    if [[ "$use_aws_iam_authenticator" == "true" ]]; then
      if [ -z "$aws_eks_cluster_name" ]; then
        echoerr 'You must specify aws_eks_cluster_name when using aws_iam_authenticator.'
        exit 1
      fi
      echo "    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - token
      - -i
      - ${aws_eks_cluster_name}
      command: aws-iam-authenticator
      env: null" >> "$KUBECONFIG"
    fi
```

**Comment by anneschuth** [comment]
Yes, it is currently running in production. I'm frankly a little surprised it works. But my assumption is that aws-iam-authenticator uses the aws IAM role that's assigned to concourse workers.

**Comment by superbrothers** [comment]
> But my assumption is that aws-iam-authenticator uses the aws IAM role that's assigned to concourse workers.

OK, I understand! 😄

File: assets/common.sh [hunk 3]
```diff
@@ -79,6 +79,26 @@ setup_kubectl() {
     exe kubectl config set-context "$CONTEXT_NAME" --user="$AUTH_NAME" --cluster="$CLUSTER_NAME"
 
     exe kubectl config use-context "$CONTEXT_NAME"
+
+    # Optional. Use the AWS EKS authenticator
+    local use_aws_iam_authenticator
```
Comment (superbrothers) [inline assets/common.sh]: I don't feel the need `use_aws_iam_authenticator` and `aws_eks_cluster_name` source params. If you'd like to use aws-iam-authenticator, you can specify so in `kubeconfig` source param:

```yaml
sources:
- name: cluster
  type: kubernetes-resource
  source:
    kubeconfig: |
      # [...]
      users:
      - name: kubernetes-admin
        user:
          exec:
            apiVersion: client.authentication.k8s.io/v1alpha1
            command: aws-iam-authenticator
            args:
              - "token"
              - "-i"
              - "((CLUSTER_ID))"
              - "-r"
              - "((ROLE_ARN))"
```

It is more flexible then specific source params.

- https://github.com/kubernetes-sigs/aws-iam-authenticator#4-set-up-kubectl-to-use-authentication-tokens-provided-by-aws-iam-authenticator-for-kubernetes

File: Dockerfile [hunk 4]
```diff
@@ -6,13 +6,27 @@ ARG KUBERNETES_VERSION=
...
+ENV GOPATH /root/.go
+ENV PATH /root/.go/bin:$PATH
+RUN cd /opt && \
+    curl -s -LO https://storage.googleapis.com/golang/go${GOVERSION}.linux-amd64.tar.gz && \
+    tar zxf go${GOVERSION}.linux-amd64.tar.gz && \
+    rm go${GOVERSION}.linux-amd64.tar.gz && \
+    ln -s /opt/go/bin/go /usr/bin/ && \
+    mkdir $GOPATH
+
+RUN go get -u -v github.com/kubernetes-sigs/aws-iam-authenticator/cmd/aws-iam-authenticator
```
Comment (superbrothers) [inline Dockerfile]: Please use multi-stage builds like:
```dockerfile
FROM golang:1.10

RUN set -x && \
    go get -u -v github.com/kubernetes-sigs/aws-iam-authenticator/cmd/aws-iam-authenticator

FROM ubuntu:16.04

...

COPY --from=0 /go/bin/aws-iam-authenticator /usr/local/bin/

RUN mkdir -p /opt/resource
COPY assets/* /opt/resource/
```

File: Dockerfile (hunk 2 above)
Comment (superbrothers) [reply Dockerfile]: By using multi-stage-builds with golang:1.10, we don't need to install git.

**Comment by superbrothers** [comment]
@anneschuth ping

**Comment by anneschuth** [comment]
Thanks for the ping, @superbrothers, we've got this running in production, it disappeared from my radar.

File: assets/common.sh (hunk 3 above)
Comment (anneschuth) [reply assets/common.sh]: There seems to be a tension between flexibility and ease of use. I think the parameters make it easier to use this: it removes the need to specify entire kubeconfigs.

**Comment by superbrothers** [comment]
@anneschuth Thank you for your contribution :blush:

**Pull request merged**